sdccc/src/main/java/com/draeger/medical/sdccc/tests/util/MdibHistorian.java

```diff
@@ -205,10 +224,30 @@ public HistorianResult episodicReportBasedHistory(final String sequenceId)
         try {
             final var messages =
                     messageStorage.getInboundMessagesByBodyType(Constants.RELEVANT_REPORT_BODIES.toArray(new QName[0]));
+            final BigInteger[] last = new BigInteger[1];
```

Comment:
You should not bypass the restriction by using such a workaround. This could lead to unexpected errors. I would suggest to use AtomicReference instead.

Reply:
After checking the different options, this turned out to be the recommended way to do this.

Reply:
However, note that this solution does not work on parallel streams. Hence I added a call to .sequential(), which ensures that the stream is not processed in parallel.